helix-core/src/selection.rs

```diff
@@ -224,6 +224,7 @@ impl Selection {
         // TODO: we could do with one vec by removing elements as we mutate
 
         for (i, range) in ranges.into_iter().enumerate() {
+            let result_len = result.len();
```

Comment:
Can we just use `result.len()` below without having to calculate this unnecessarily?

Reply:
No, result is borrowed below, so len() is unavailable there.